### PR TITLE
Feature/payment option

### PR DESCRIPTION
**Pull request by Ponchimeow**
+ add PaymentInput, when paymentType is perpetual enable

+ placeOrder, paymentModel add paymentMethod

+ payment option value: CREDIT, VACC, CVS, InstFlag, UNIONPAY, WEBATM, BARCODE

+ hasura app_setting add
    - payment.method.credit.enable, 1 enable, 0 disable
    - payment.method.vacc.enable, 1 enable, 0 disable
    - payment.method.cvs.enable, 1 enable, 0 disable
    - payment.method.instflag.enable, 1 enable, 0 disable
    - payment.method.unionpay.enable, 1 enable, 0 disable
    - payment.method.barcode.enable, 1 enable, 0 disable

### CODE REVIEW

File: src/components/checkout/PaymentInput.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+const PaymentInput: React.FC<{
+  value: PaymentMethodType
+  onChange: (value: PaymentMethodType) => void
+}> = ({ value, onChange }) => {
+  const { formatMessage } = useIntl()
+  const { settings } = useApp()
+  const [selectedPaymentMethodType, setSelectedPaymentMethodType] = useState<PaymentMethodType | null>(value)
+
+  const handleChange: (paymentMethod?: PaymentMethodType | null) => void = PaymentMethodType => {
```
Comment (futianshen) [inline src/components/checkout/PaymentInput.tsx:29]: just use type inference
```ts
const handleChange = (PaymentMethodType: PaymentMethodType | null) => {
```

File: src/components/checkout/PaymentInput.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+`
+const StyledDescription = styled.div`
+  color: var(--gray-dark);
+  font-size: 14px;
+  letter-spacing: 0.4px;
+`
+
+export type PaymentMethodType = 'CREDIT' | 'VACC' | 'CVS' | 'InstFlag' | 'UNIONPAY' | 'WEBATM' | 'BARCODE'
+
+const PaymentInput: React.FC<{
```
Comment (futianshen) [inline src/components/checkout/PaymentInput.tsx:21]: `PaymentMethodSelector` is a better name

File: src/components/checkout/PaymentInput.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  const { formatMessage } = useIntl()
+  const { settings } = useApp()
+  const [selectedPaymentMethodType, setSelectedPaymentMethodType] = useState<PaymentMethodType | null>(value)
+
+  const handleChange: (paymentMethod?: PaymentMethodType | null) => void = PaymentMethodType => {
+    const currentPaymentOption =
+      typeof PaymentMethodType === 'undefined' ? selectedPaymentMethodType : PaymentMethodType
+    typeof PaymentMethodType !== 'undefined' && setSelectedPaymentMethodType(PaymentMethodType)
+    currentPaymentOption && localStorage.setItem('kolable.cart.paymentMethod', currentPaymentOption)
+    currentPaymentOption && onChange && onChange(currentPaymentOption)
```
Comment (futianshen) [inline src/components/checkout/PaymentInput.tsx:34]: use if statement is more readable

File: src/components/checkout/PaymentSelector.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        )}
+        {Number(settings['payment.spgateway.unionpay.enable']) && (
+          <Select.Option value="UNIONPAY">{formatMessage(checkoutMessages.label.unionPay)}</Select.Option>
+        )}
+        {Number(settings['payment.spgateway.webatm.enable']) && (
+          <Select.Option value="WEBATM">{formatMessage(checkoutMessages.label.webAtm)}</Select.Option>
+        )}
+        {Number(settings['payment.spgateway.barcode.enable']) && (
+          <Select.Option value="BARCODE">{formatMessage(checkoutMessages.label.barcode)}</Select.Option>
+        )}
```
Comment (Elantris) [inline src/components/checkout/PaymentSelector.tsx:70]: to be more readable and secure, the condition should be `settings['payment.spgateway.credit.enable'] === '1'`

**Pull request closed without merging**